server/src/main/java/org/opensearch/wlm/SearchWorkloadTransportHandler.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.wlm;
+
+import org.opensearch.search.fetch.ShardFetchRequest;
+import org.opensearch.search.internal.InternalScrollSearchRequest;
+import org.opensearch.search.internal.ShardSearchRequest;
+import org.opensearch.search.query.QuerySearchRequest;
+import org.opensearch.tasks.Task;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportChannel;
+import org.opensearch.transport.TransportRequest;
+import org.opensearch.transport.TransportRequestHandler;
+
+/**
+ * This class is mainly used to populate the queryGroupId header
+ * @param <T> T is Search related request
+ */
+public class SearchWorkloadTransportHandler<T extends TransportRequest> implements TransportRequestHandler<T> {
+
+    private final ThreadPool threadPool;
+    TransportRequestHandler<T> actualHandler;
```

Comment:
This class can be internal to SearchWorkloadTransportInterceptor?

Reply:
Makes sense since the class is not used anywhere else.

Reply:
We can keep the inner class public. Atleast, we don't pollute the namespace that way?

Reply:
Also, should we rename this class to `WorkloadManagementTransportRequestHandler`?